plugins/oracle/identity/adminUserAPIKeys.js

```diff
@@ -0,0 +1,105 @@
+    recommended_action: 'Remove API keys for administrator users',
+    apis: ['user:list', 'apiKey:list', 'group:list', 'userGroupMembership:list', 'authToken:list'],
+
+    run: function (cache, settings, callback) {
+        var results = [];
+        var source = {};
+
+        var region = helpers.objectFirstKey(cache['regionSubscription']['list']);
+
+        var users = helpers.addSource(cache, source,
+            ['user', 'list', region]);
+
+        var groups = helpers.addSource(cache, source,
+            ['group', 'list', region]);
+
+        var userGroups = helpers.addSource(cache, source,
+            ['userGroupMembership', 'list', region]);
+
+        var apiKeys = helpers.addSource(cache, source,
+            ['apiKey', 'list', region]);
+
+        if (!users || !groups || !userGroups || !apiKeys) return callback(null, results, source);
+
+        if (users.err || !users.data) {
+            helpers.addResult(results, 3,
+                'Unable to query for users: ' + helpers.addError(users), 'global');
+            return callback(null, results, source);
+        }
+
+        if (!users.data.length) {
+            helpers.addResult(results, 0, 'No user accounts found', 'global');
+            return callback(null, results, source);
+        }
+
+        if (groups.err || !groups.data) {
+            helpers.addResult(results, 3,
+                'Unable to query user groups: ' + helpers.addError(groups), 'global');
+            return callback(null, results, source);
+        }
+
+        if (!groups.data.length) {
+            helpers.addResult(results, 0, 'No groups found', 'global');
+            return callback(null, results, source);
+        }
+
+        if (userGroups.err || !userGroups.data) {
+            helpers.addResult(results, 3,
+                'Unable to query user groups: ' + helpers.addError(userGroups), 'global');
+            return callback(null, results, source);
+        }
+
+        if (!userGroups.data.length) {
+            helpers.addResult(results, 0, 'No user group membership found', 'global');
+            return callback(null, results, source);
+        }
+
+        if (apiKeys.err || !apiKeys.data) {
+            helpers.addResult(results, 3,
+                'Unable to query user API keys: ' + helpers.addError(apiKeys), 'global');
+            return callback(null, results, source);
+        }
+
+        if (!apiKeys.data.length) {
+            helpers.addResult(results, 0, 'No user API keys found', 'global');
+            return callback(null, results, source);
+        }
+
+        let adminGroup = groups.data.find(group => group.name == 'Administrators');
+        let apiKeyIds = apiKeys.data.map(apiKey => apiKey.userId) || [];
+
+        if (adminGroup) {
+            adminGroupId = adminGroup.id;
+
+            let adminUsers = userGroups.data.map(userGroup => userGroup.userId) || [];
```

Comment:
This api data will always give userid of admin users? no need to check for any flag or something?

Reply:
If it doesn't, userGroup.userId will be undefined but code won't break